libpromises/evalfunction.c

```diff
@@ -5276,38 +5276,21 @@ static FnCallResult FnCallFold(EvalContext *ctx,
 
 /*********************************************************************/
 
-static FnCallResult FnCallDatatype(EvalContext *ctx, ARG_UNUSED const Policy *policy, const FnCall *fp, const Rlist *finalargs)
+static const char * DataTypeStringFromVarName(EvalContext *ctx, const char *var_name, bool detail)
```

Comment:
`const char *` commonly means / implies a string owned by something else, or in other words a string you don't need to free.